### ISSUE TITLE
Feature: Auto-generate Deprecation Timeline/Table from Wrapper Usage

### ISSUE THREAD

**Issue by Borda** [Borda/pyDeprecate#88]
### Is your feature request related to a problem? Please describe.
Currently, tracking the lifecycle of deprecated features requires manual effort. Developers often have to keep a separate "Deprecation Schedule" in the documentation or CHANGELOG, which easily falls out of sync with the actual code.

There is no automated way to visualize:
1.  **What** is currently deprecated.
2.  **When** it was introduced (deprecated version).
3.  **When** it will be deleted (removal version).

### Describe the solution you'd like
I would like a feature that automatically scans the codebase, detects usages of the `@deprecated` wrapper, and scrapes the arguments (e.g., `version`, `delete_version`) to generate a summary.

This ensures the **code is the single source of truth**.

The tool should output the data in two formats:

**1. Markdown Table (for static docs)**
A summary table showing the version matrix.

**2. Mermaid.js Timeline (for visual roadmaps)**
A chart definition that illustrates the deprecation windows.

#### 1. Markdown Table Output
Generated by scraping the wrappers:

| Symbol | Deprecated In | Removal Target | Current Status |
| :--- | :---: | :---: | :--- |
| `api.old_method` | v1.2.0 | v2.0.0 | ⚠️ Active Warning |
| `legacy_module` | v1.5.0 | v2.0.0 | ⚠️ Active Warning |
| `utils.helper` | v1.0.0 | v1.8.0 | ❌ Past Removal Date |

#### 2. Mermaid Timeline Output
Generated for visual representation:

'''mermaid
gantt
    title Deprecation Lifecycle
    dateFormat  X
    axisFormat %s

    section API
    api.old_method (v1.2 -> v2.0)  :crit, active, v1.2, v2.0
    legacy_module (v1.5 -> v2.0)   :crit, active, v1.5, v2.0

    section Utils
    utils.helper (v1.0 -> v1.8)    :done, v1.0, v1.8
'''

### Describe alternatives you've considered
* **Manual Tracking:** Manually writing tables in README.md, which is error-prone.
* **Search Tools:** Using `grep` or IDE search to find `@deprecated`, which is unstructured and hard to read at a glance.

### Implementation hints
The implementation should leverage the existing developer utilities found in `pyDeprecate` to ensure consistency and reuse logic.

Specifically, the **`validate_deprecated_callable`** utility (and the `DeprecatedCallableInfo` dataclass) already implements the logic for:
1.  Inspecting a callable.
2.  Extracting the `__deprecated__` attribute.
3.  Validating the configuration (checking for zero-impact wrappers, etc.).

We can extend this pattern to:
1.  Walk the package modules/classes.
2.  Pass discovered callables to `validate_deprecated_callable` (or a similar inspection helper) to extract the `deprecated_in` and `remove_in` versions.
3.  Aggregate this data to build the Markdown table or Mermaid chart.

**Comment by Borda** [comment]
Building on this — two extra columns would make the table significantly more actionable:

**1. Item type** — what is being deprecated:

- `function` / `method` — decorated with `@deprecated`
- `class` — wrapped via `deprecated_class()`
- `argument` — `TargetMode.ARGS_REMAP` with `args_mapping` (parameter rename, not callable removal)

**2. Deprecation mode** — what happens at call time, derivable from `DeprecationConfig.target`:

| `target` value                           | Mode label            | Meaning                                         |
| ---------------------------------------- | --------------------- | ----------------------------------------------- |
| `TargetMode.NOTIFY`                      | `warn-only`           | emits warning, executes original body unchanged |
| callable                                 | `redirect → new.path` | forwards call to new function/class             |
| `TargetMode.ARGS_REMAP` + `args_mapping` | `arg-rename`          | remaps kwargs, runs original body               |

Proposed extended table:

| Symbol           | Type     | Mode                         | Deprecated in | Removal target | Status     |
| ---------------- | -------- | ---------------------------- | ------------- | -------------- | ---------- |
| `api.old_method` | function | `redirect → api.new_method`  | v1.2.0        | v2.0.0         | ⚠ active   |
| `Model`          | class    | `redirect → NewModel`        | v1.5.0        | v2.0.0         | ⚠ active   |
| `train(lr=...)`  | argument | `arg-rename → learning_rate` | v1.3.0        | v1.8.0         | ❌ overdue |
| `legacy_fn`      | function | `warn-only`                  | v1.0.0        | —              | ⚠ active   |

All fields already exist in `DeprecationWrapperInfo` + `DeprecationConfig` — no schema changes needed, pure rendering layer work.

**Comment by Borda** [comment]
for this, we may also need to add module-level deprecation...
